libmscore/excerpt.cpp

```diff
@@ -1154,22 +1154,26 @@ QList<Excerpt*> Excerpt::createAllExcerpt(MasterScore *score)
 //   createName
 //---------------------------------------------------------
 
-QString Excerpt::createName(const QString& partName, QList<Excerpt*> excerptList)
+QString Excerpt::createName(const QString& partName, QList<Excerpt*>& excerptList)
       {
-      QString n = partName.simplified();
-      QString name;
-      int count = excerptList.count();
-      for (int i = 0;; ++i) {
-            name = i ? QString("%1-%2").arg(n).arg(i) : QString("%1").arg(n);
-            Excerpt* ee = 0;
-            for (int k = 0; k < count; ++k) {
-                  ee = excerptList[k];
-                  if (ee->title() == name)
-                        break;
-                  }
-            if ((ee == 0) || (ee->title() != name))
+      QString name = partName.simplified();
+      int count = 0;    // no of occurences of partName
+      
+      foreach(Excerpt* e, excerptList) {
+            // if <partName> already exists, change <partName> to <partName 1>
+            if (e->title().compare(name) == 0) {
+                  e->setTitle(e->title() + QString(" %1").arg("1"));
+                  count = 1;
                   break;
+                  }
+          
+            if (e->title().mid(0, e->title().size() - 2) == name)
```

Comment:
Works for any number of parts now :)
<img width="698" alt="screen shot 2017-02-16 at 10 11 12 pm" src="https://cloud.githubusercontent.com/assets/14026308/23030837/fb96781a-f494-11e6-8ea3-30f993ad1f7b.png">